test/test_implicit_free_surface_solver.jl

```diff
@@ -29,31 +30,15 @@ function set_simple_divergent_velocity!(model)
     v .= 0
     η .= 0
 
-    imid = Int(floor(grid.Nx / 2)) + 1
-    jmid = Int(floor(grid.Ny / 2)) + 1
-
-    i, j, k = imid, jmid, 1
-
-    if grid isa RectilinearGrid
-        Δx, Δy = grid.Δxᶠᵃᵃ, grid.Δyᵃᶜᵃ
-    end
-
-    if grid isa ImmersedBoundaryGrid && grid.underlying_grid isa RectilinearGrid
-        Δx, Δy = grid.underlying_grid.Δxᶠᵃᵃ, grid.underlying_grid.Δyᵃᶜᵃ
-    end
+    # pick a surface cell at the middle of the domain
+    i, j, k = Int(floor(grid.Nx / 2)) + 1, Int(floor(grid.Ny / 2)) + 1, grid.Nz
+    inactive_cell(i, j, k, grid) && error("The nudged cell at ($i, $j, $k) is inactive.")
 
-    if grid isa LatitudeLongitudeGrid
-        Δx, Δy = CUDA.@allowscalar grid.Δxᶠᶜᵃ[i], grid.Δyᶜᶠᵃ
-    end
-    if grid isa ImmersedBoundaryGrid && grid.underlying_grid isa LatitudeLongitudeGrid
-        Δx, Δy = CUDA.@allowscalar grid.underlying_grid.Δxᶠᶜᵃ[i], grid.underlying_grid.Δyᶜᶠᵃ
-    end
+    Δy = Δyᶜᶠᶜ(i, j, k, grid)
+    Δz = Δzᶜᶠᶜ(i, j, k, grid)
```

Comment:
and these?

Reply:
oh these might be better... I didn't know about these: `Δyᶜᶠᶜ`, `Δzᶜᶠᶜ`. Don't these require `CUDA.@allowscalar`?

Reply:
Hmmm they should... I am not sure why they don't error